### ISSUE TITLE
Separate tooling pages

### ISSUE THREAD

**Issue by mxstbr** [styled-components/styled-components-website#88]
Since the tooling docs are quite long, they don't really make sense to all live on a single huge page. I think it'd make sense to move them all to actually separate pages:

- `/docs/tooling/babel-plugin`
- `/docs/tooling/stylelint`
- etc.

/cc @philpl wdyt?

**Comment by philpl** [comment]
I think it might be a slippery slope of follow up problems. Suddenly you are dealing with a change on every structure, also in typography, nested menus, etc.

I’m not confident enough yet to think about this change but it will definitely become more urgent. For now it should hold up ok for the time being as the navigation has enough anchors. But we will need to implement separate pages soon, I agree

**Comment by mxstbr** [comment]
Well the menu would stay the same, it's just that the subanchors would link to pages and the main anchor would show a list of subpages with an overview or something?

**Comment by philpl** [comment]
@mxstbr I think it could work but frankly, it's tricky to pull off. You're disrupting the flow of the page, and you'll need an index page for them. I think we can do that, but we'll need to test it before we ship it, as I'm unsure whether it'll work well :/

**Comment by mxstbr** [comment]
The flow of the page? It's three entirely separate tools put together on one massive page, there's no flow to that right now either!

**Comment by philpl** [comment]
@mxstbr I’m referring to these sections being part of a page. Once we create sub pages we need to think about switching between them. But the NextPage component could actually help with that?

**Comment by mxstbr** [comment]
I'm talking in terms of the layout as a user, not in terms of the code; no idea how the code is set up rn.

To me, having the Babel plugin, the stylelint processor, the Jest matchers all on one page is suuuper weird and hard to navigate. Those should just be their own pages where these tools can live.

**Comment by philpl** [comment]
@mxstbr yea, I agree that we need to separate them, but also trying to bring up the problems around it :) there just seem to be a lot of questions that need to be answered before we can jump onto implementing it

**Comment by mxstbr** [comment]
> Once we create sub pages we need to think about switching between them.

Well I mean that's what the breadcrumbs in the sidebar are for, right? That's why we show all the subsections there, so you can jump between them!

**Comment by philpl** [comment]
Yes, but we’ll need to think about switching to the next page vs the next “topic” I.e. basics → advanced. We can change the NextPage component to display the next page instead of just the next “topic”. That would solve all problems imho 

Having just the option to switch to the next page in the sidebar isn’t enough imho, as it prevents people from just reading on, as they’d have to look at the sidebar first

**Comment by mxstbr** [comment]
Ahhh yeah sure, that's a code thing though. Agreed.

**Comment by philpl** [comment]
@mxstbr Well, I was referring to the UX problem, but yea, should we open a separate issue first to collect ideas, or rename this one? I think it already developed into a proper discussion 😆

**Comment by mxstbr** [comment]
Let's keep this one about tooling, the NextPage component behaviour wouldn't actually change compared to how it works today until we add the tooling pages so it's more of a refactor than a UX change?
